client/src/components/ExpandedRowComponent.tsx

```diff
@@ -0,0 +1,128 @@
+
+export const useExpandedRowComponent = ({onUpdate, user}: {onUpdate: CallableFunction; user?: {displayName: string}}) => {
+    const ExpandedRowComponent = ({data}: {data: CodeReview}) => {
+        const handleActionClick = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            const action = currentTarget.getAttribute('name');
+            const value = currentTarget.getAttribute('value');
+
+            fetchData(`/api/action/${action}/${value}`)
+                .then((result) => {
+                    result?.data && onUpdate && onUpdate({
+                        data: result.data,
+                        action,
+                    });
+                })
+                .catch((e) => {
+                    setErrorMessage('Error saving data, see console error log for details.');
+                    logError(`Error sending: /api/action/${action}/${value}`, e);
+                })
+        };
+        const ActionButton = ({label, reviewId}: {label: string; reviewId: number}) => {
+            const sanitizedName = label.toLowerCase().replace(' ', '-');
+
+            return (
+                <button
+                    id={`${sanitizedName}-${reviewId}`}
+                    name={sanitizedName}
+                    value={reviewId}
+                    onClick={handleActionClick}
+                >{label}</button>
+            );
+        };
+        const showClaim = !user || !data.reviewers?.includes(user?.displayName);
+        const showApprove = !user || !data.approvers?.includes(user?.displayName);
+        const showChange = !user || !data.requestChanges?.includes(user?.displayName);
+        const [editing, setEditing] = useState(false);
+        const [errorMessage, setErrorMessage] = useState(null as null | string);
+        const [saveClicked, setSaveClicked] = useState(false);
+        const dirtyData = {...data};
+
+        const handleSaveClicked = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+            if (JSON.stringify(dirtyData) !== JSON.stringify(data)) {
+                const value = currentTarget.getAttribute('value');
+
+                setSaveClicked(true);
+                fetchData(`/api/action/save/${value}`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        note: dirtyData.note,
+                        jiraTicket: dirtyData.jiraTicket,
+                    })
+                })
+                    .then((result) => {
+                        if (!result.data) {
+                            setErrorMessage('Error saving data, see console error log for details.');
+                            logError(`Error saving: /api/action/save/${value}`, result);
+                            return;
+                        }
+                        Object.assign(data, result.data);
```

Comment:
this is a no-no -- this component should not be changing the value of its prop. It should be passed a function from the parent that calls it which would allow it to update the prop, e.g.
`handleDataUpdate(newData)` which would change the value of `data`

Reply:
Good catch, this was a debug code that didn't get clean up :facepalm: